.github/workflows/ci.yml

```diff
@@ -132,6 +132,20 @@ jobs:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # pin v4.1.7
         timeout-minutes: 5
 
+      - name: Ensure the PR head ref is not a perennial branch
+        if: github.event_name == 'pull_request'
+        run: |
+          if [[ "${{ github.head_ref }}" =~ ^releases/.*$ ]]; then
+            (
+              echo 'You are using a perennial branch as a PR head ref, this is forbidden because once merge the branch will be deleted by GitHub.';
+              echo 'If you want to create a PR with the change of a perennial branch, create a feature branch from it (git switch --create my-feature-branch $PERENNIAL_BRANCH)';
```

Comment:
```suggestion
              echo 'You are using a perennial branch as a PR head ref, this is forbidden because once merged the branch will be deleted by GitHub.';
              echo 'If you want to create a PR with a change from a perennial branch, create a feature branch from it (git switch --create my-feature-branch $PERENNIAL_BRANCH)' and use the feature branch as the PR head ref;
```